src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -1321,6 +1321,14 @@ private BadHttpRequestException GetInvalidRequestTargetException(ReadOnlySpan<by
                 ? target.GetAsciiStringEscaped(Constants.MaxExceptionDetailSize)
                 : string.Empty);
 
+    public void DisableHttp1KeepAlive()
```

Comment:
I should have asked for a comment earlier to make it clearer what the purpose of this is.

```suggestion
    // This is called during certain bad requests so the automatic Connection: close header gets sent with custom responses.
    // If no response is written, SetBadRequestState(BadHttpRequestException) will later also modify the status code.
    public void DisableHttp1KeepAlive()
```